firedrake/slate/preconditioners.py

```diff
@@ -170,9 +171,9 @@ def initialize(self, pc):
             if "on_boundary" in neumann_subdomains:
                 measures.append(ds)
             else:
-                measures.append(ds(tuple(neumann_subdomains)))
+                measures.extend([ds(sd) for sd in neumann_subdomains])
                 dirichlet_subdomains = set(mesh.exterior_facets.unique_markers) - neumann_subdomains
-                trace_subdomains.append(sorted(dirichlet_subdomains))
+                trace_subdomains.extend([sd for sd in dirichlet_subdomains])
```

Comment:
You lost the sorting, I think that is important.  So just write: `trace_subdomains.extend(sorted(dirichlet_subdomains))`